code/game/dna/dna_modifier.dm

```diff
@@ -138,10 +138,16 @@
 			M.forceMove(get_turf(src))
 
 /obj/machinery/dna_scannernew/update_icon_state()
-	if(occupant)
-		icon_state = "scanner_occupied"
-	else
+	if(panel_open)
+		icon_state = "scanner" + (occupant ? "" : "_open") + "_maintenance"
```

Comment:
```suggestion
		icon_state = "scanner_[occupant ? "" : "open_"]maintenance"
```
We usually do something like this, not sure if we have any guidelines on it currently
Though it does not look extremely nice, fully up to you